### DISCUSSION TITLE
Can the Rosetta 2 caveat be hidden when running on Intel Mac?

### DISCUSSION THREAD

**Discussion by jorikcaljouw** [Homebrew/discussions#5518] [Casks] [answered]
### Output of `brew config`

```shell
❯ brew config
HOMEBREW_VERSION: 4.3.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c3ae0327c01aab461531bfc0fcdfb4dd3a779205
Last commit: 5 days ago
Core tap JSON: 23 Jul 21:36 UTC
Core cask tap JSON: 23 Jul 21:36 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.kVMDnJXCPv/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.45.2 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 13.6.7-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
```

### Output of `brew doctor`

```shell
❯ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  cakebrew
  meld

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
Untap them with `brew untap`.

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

Just like in https://github.com/orgs/Homebrew/discussions/5396, when updating or installing mkvtoolnix cask, you see the following message (also on an intel mac):
```
==> mkvtoolnix
mkvtoolnix is built for Intel macOS and so requires Rosetta 2 to be installed.
You can install Rosetta 2 with:
  softwareupdate --install-rosetta --agree-to-license
Note that it is very difficult to remove Rosetta 2 once it is installed.
```

I believe it is confusing to see a message that Rosetta 2 is required when in fact it is not! Even if you think all homebrew users should know what Rosetta 2 is and that it would be ridiculous to try to install it on an intel mac, the message is wrong without specifying it is only required when using macOS on Apple Silicon. If this cannot be suppressed when on intel mac, then I think the message itself should just be extended from: 
`$cask is built for Intel macOS and so requires Rosetta 2 to be installed.`
to
`$cask is built for Intel macOS and so requires Rosetta 2 to be installed when used on Apple Silicon macOS.`
or something like that.

What is your opinion about this?

### Relevant casks

mkvtoolnix

**Comment by carlocab** [comment, marked as answer]
You shouldn't be seeing that caveat on an Intel Mac. That's a bug; can you report it at https://github.com/Homebrew/brew/issues/new?assignees=&labels=bug&projects=&template=bug.yml?

**Reply by jorikcaljouw** [reply]
Thanks for the quick reply, just created [#17839](https://github.com/Homebrew/brew/issues/17839)
